### PR TITLE
Ensure go.mod matches apis/go.mod

### PR DESCRIPTION
**Pull request by 2uasimojo**
Add a utility that compares statements in go.mod and apis/go.mod. If
matching module paths are found, the utility determines whether the
versions match, printing a human-readable report and exiting nonzero iff
*any* versions *do not* match.

Add a make rule, `make modcheck`, to run it.

Plumb `modcheck` into `verify`.

[HIVE-1941](https://issues.redhat.com//browse/HIVE-1941)

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Expect this to fail `verify` on the new check. A subsequent commit in this PR will have to sync up the versions.

File: hack/modsync.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+	return ret
+}
+
+func cmp1(root, apis map[string]string) bool {
+	insync := true
+	for path, rootver := range root {
+		apisver, ok := apis[path]
+		if !ok {
+			fmt.Printf("\t(path in root but not apis: %s)\n", path)
```
Comment (2uasimojo) [inline hack/modsync.go:91]: This can probably go away -- too noisy.

File: hack/modsync.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	}
+	return insync
+}
+
+func cmp2(root, apis map[string]replacement) bool {
+	insync := true
+	for path, rootrepl := range root {
+		apisrepl, ok := apis[path]
+		if !ok {
+			fmt.Printf("\t(path in root but not apis: %s)\n", path)
```
Comment (2uasimojo) [inline hack/modsync.go:109]: ditto

**Review by 2uasimojo — COMMENTED** [review]
docs?
CLI options like `--help`, verbosity?

**Comment by abutcher** [comment]
:+1: Very cool. We have a [dependency management section in our development docs](https://github.com/openshift/hive/blob/master/docs/developing.md#dependency-management) which would probably be an okay place for docs if needed.

**Comment by 2uasimojo** [comment]
> We have a [dependency management section in our development docs](https://github.com/openshift/hive/blob/master/docs/developing.md#dependency-management) which would probably be an okay place for docs if needed.

Good call, done.

Added a second commit to sync the modules. This should be ready for review now.

/assign @abutcher

**Pull request merged**